hyperjumptech/bookkeeping
# Refresh Go Version and Libraries

# Hello, Merge/Pull Request (PR)  

## PR type  



### what issue this pr address

1. Update golang to 1.22
2. update deprecated libs ioutils,change to io
3. fixed formatting/linter feedback by staticcheck
4. change domain from wallet to bookkeeping, update all imports/names
5. added cron worker for possible routine backup
6. added firebase lib for possible backup to firebase
7. added db dump functionality to support no 5 and 6 above

### how do you test

1. `make test` will test unit test (short version)
2. `make lint` will perform static checking
3. `make build` and then call `./bookkeeping-img-go` to run locally (ensure database is set)
